### ISSUE TITLE
Underscore in Variable Names

### ISSUE THREAD

**Issue by tk4arts** [w4tsn/node-red-contrib-protobuf#29]
Hello there,

for a project I need to encode a message out of Node-RED using a protobuf file. 

My .proto file looks something like this:
``` proto
syntax = "proto3";

import "google/protobuf/wrappers.proto";

message Department {
    google.protobuf.Int32Value department_id = 100;
    google.protobuf.StringValue name = 200;
}
```

The message I want to send looks like this:
``` js
msg.payload = {
    department_id: 12,
    name: "Name"
}
```

There are two things that are not working for me. 

1. The first one is that the import of the google/protobuf/wrappers.proto is not working. So I get the following error Message:
![image](https://user-images.githubusercontent.com/84267222/141792804-aa577d8d-e9ca-4461-ba6b-d60e23114cb6.png)

I managed to fix this issue by using the base types int32 and string. But it would be nice to use the import because then I do not have to rewrite the whole proto file.

2. If one part of the message has an underscore in it (like the department_id has) this part is ignored in the encode. So it basically is not there when encoding and decoding again.
This is the message I feed into the encoding node:
![image](https://user-images.githubusercontent.com/84267222/141793730-5d3d159c-0a4d-4367-9a6e-3725719e3788.png)
And this is the message after an encode and decode:
![image](https://user-images.githubusercontent.com/84267222/141793835-cdabbfe5-f37d-4062-8beb-636862ad41f9.png)

Unfortunately I am not able to change the naming in the proto file because it is a bigger project. 

I found another Issue online [here](https://github.com/scalapb/ScalaPB/issues/170) that also states a problem with underscores in names.

My hope is that someone of you is able to help with these two errors (but mainly with the second one).

Thanks in advance for your highly appreciated reply.

Till

**Comment by w4tsn** [comment]
Hey there Till, I'll have a look into the second problem this week

**Comment by tk4arts** [comment]
Thanks for the fast answer. 
Did you managed to have a look at the problem and figure out a solution. Because I am still stuck with this one.

Till

**Comment by w4tsn** [comment]
Hey there! I did not have enough time to test it but I think for your problem 1) it would help to either A) bundle your protobuf files into one file or import your dependencies too by entering comma-separated file paths.

For your second problem I'll have to dig a bit deeper, couldn't manage this week.

**Comment by tk4arts** [comment]
As you closed the issue. Did you find a solution to the problem with the underscores in the variable names?

**Comment by w4tsn** [comment]
@tk4arts hey there! Yes I actually did ;) Please test it with v1.1.1. Sorry I didn't notify you earlier, somehow assumed that the commit reference used to close that issue already answers that question and notifies you :)

**Comment by tk4arts** [comment]
Hey thanks for the quick reply. I missed the commit as in node red the most recent version showed was still 1.1.0.

With 1.1.1 the underscore problem is fixed thank you very much!
Regarding the other issue I am not quite sure what you mean with comma seperated file paths.

Could you make an example with the code above?

Thanks in advance!

**Comment by w4tsn** [comment]
In the protofile node it is possible to supply multiple paths comma-separated. So instead of just putting in the path to your proto file(s) you also enter the paths to any files you want to import so they get loaded into the protofile node together. Then it should be possible to actually reference and import types from other files etc.

So in the protofile path field you'd enter

```
path/to/your/file.proto,google/protobuf/wrappers.proto
```

Instead of only

```
path/to/your/file.proto
```

**Comment by tk4arts** [comment]
I just tried your suggestion and I am still getting the error that the message is not valid under selected message type. I will try it again tomorrow.

**Comment by tk4arts** [comment]
Ok I tried it again with the same result that it is not working.
I will try to describe what I did in case I made some sort of stupid mistake.

My folder Structure from the root node-red directory is:

```
1. PROTO
    - my-proto-file.proto
2. google
    a. protobuf
        - wrappers.proto
```

So now to import the two files in the protofile path I did the following:
```
PROTO/my-proto-file.proto,google/protobuf/wrappers.proto
```

My protofile and message looks the same like in the first post.

And I still get the error:
![image](https://user-images.githubusercontent.com/84267222/142989568-93189093-89eb-47c1-b31e-617e607ff2ac.png)

Maybe I am missing something with the import and the use of the file. 

``` proto
syntax = "proto3";

import "google/protobuf/wrappers.proto";

message Department {
    google.protobuf.Int32Value department_id = 100;
    google.protobuf.StringValue name = 200;
}
```

When I specify the path of the file in the protofile node how do I import the file into another proto file?

I hope it will be possible.  The underscore worked like a charm thank you very much for the quick response and fix!

Regards
Till

**Comment by w4tsn** [comment]
Have you tried not importing the file at all? It could be, since the import is done initially for all the files at once, that the types are already present and you don't need an import here. Still it would be nice to support the import statement so one does not necessarily have to rewrite their files.

As a work-around we could remove import statements from the loaded files or something, if it turns out that those are incompatible with the way we load proto files in the node.

Please also open another issue for this, since it's clearly a separate issue from the underscore one :)

**Comment by vdwpsmt** [comment]
hey, 

I'm using version 1.1.1 and also encountered the problem with the underscore in the variable name.  Any updates on this?
thanks

**Comment by w4tsn** [comment]
@vdwpsmt please open another issue detailing what your problem is exactly and how to reproduce this. To my knowledge this issue is fixed in 1.1.1.

Did you enable the check-box in the configuration node for "keep snake case"?

**Comment by vdwpsmt** [comment]
@w4tsn oeps.  I wasn't aware of the check-box "keep snake case".  Works fine with this checkbox enabled.  Sorry for the confusion.
